docs/rfc/active/business_glossary/README.md

```diff
@@ -0,0 +1,293 @@
+```
+
+A Dataset Field URN(DatasetFieldUrn will be like below (this is being added as part of another RFC)
+```
+urn:li:datasetField:(<datasetUrn>,<fieldPath>)
+```
+
+### New Snapshot Object
+There will be new snapshot object will be definied to onboard business terms along with definitions
+
+Path : metadata-models/src/main/pegasus/com/linkedin/metadata/snapshot/
+```
+/**
+ * A metadata snapshot for a specific BusinessTerm entity.
+ */
+record BusinessTermSnapshot {
+
+  /**
+   * URN for the entity the metadata snapshot is associated with.
+   */
+  urn: BusinessTermUrn
+
+  /**
+   * The list of metadata aspects associated with the dataset. Depending on the use case, this can either be all, or a selection, of supported aspects.
+   */
+  aspects: array[BusinessTermAspect]
+}
+```
+
+Path : metadata-models/src/main/pegasus/com/linkedin/metadata/aspect/
+
+### BusinessTermAspect
+There will be new aspect defined to capture the required attributes & ownership information
+
+```
+/**
+ * A union of all supported metadata aspects for a BusinessTerm
+ */
+typeref BusinessTermAspect = union[
+  BusinessTermProperties,
+  Ownership
+]
+```
+
+Business Term Entity Definition
+```
+/**
+ * Data model for a Business Term entity
+ */
+record BusinessTermEntity includes BaseEntity {
+
+  /**
+   * Urn for the dataset
+   */
+  urn: BusinessTermUrn
+
+  /**
+   * Business Term native name e.g. CashInstrument
+   */
+  name: optional string
+
+}
+```
+
+### Entity BusinessTermProperties
+
+```
+/**
+ * Properties associated with a BusinessTerm
+ */
+record BusinessTermProperties {
+
+  /**
+   * Friendly Name of business term
+   */
+  name: string
+
+  /**
+   * Definition of business term
+   */
+  definition: string
+
+  /**
+   * Source of the Business Term (INTERNAL or EXTERNAL) with default value as INTERNAL
+   */
+  termSource: EnumType
+
+  /**
+   * External Reference to the business-term (URL)
+   */
+  sourceRef: optional string
+
+ /**
+   * URI of the external source when the term is borrowed from external
+   * e.g: fibo-fbc-fi-fi:CashInstrument
+   */
+  sourceURI: optional string
+
+  /**
+   * The abstracted URI such as https://spec.edmcouncil.org/fibo/ontology/FBC/FinancialInstruments/FinancialInstruments/CashInstrument.
+   */
+  uri: optional Uri
+
+}
+
+```
+
+### Business Term Realationship 
+Business Terms will be owened by certain business users
+
+```
+/**
+ * A generic model for the Owned-By relationship
+ */
+@pairings = [ {
+  "destination" : "com.linkedin.common.urn.CorpuserUrn",
+  "source" : "com.linkedin.common.urn.BusinessTermUrn"
+}, {
+   "destination" : "com.linkedin.common.urn.CorpuserUrn",
+   "source" : "com.linkedin.common.urn.BusinessTermUrn"
+} ]
+record OwnedBy includes BaseRelationship {
+
+  /**
+   * The type of the ownership
+   */
+  type: OwnershipType
+}
+```
+
+### Extension to SchemaField entity
+
+Proposing to replace the fieldPath with urn
+
+```
+record SchemaField {
```

Comment:
I believe even a dataset can have business glossary. 

So, think of creating `BusinessGlossary` aspect like below and add this model as aspect of Dataset and DatasetField ?  


```pdl
record BusinessGlossary{
   businessGlossaryUrn : BusinessGlossaryUrn, 
   createdBy: ActorUrn
}
```

Reply:
Introduced BusinessTerm aspect and extended the DatasetAspect model to include BusinessTerm and also did the similar for DatasetField